catroid/src/androidTest/java/org/catrobat/catroid/test/io/asynctask/ProjectExporterTest.kt

```diff
@@ -0,0 +1,142 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.io.asynctask
+
+import android.Manifest
+import android.app.NotificationManager
+import android.content.Context
+import android.util.Log
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.GrantPermissionRule
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.common.Constants
+import org.catrobat.catroid.content.Project
+import org.catrobat.catroid.io.StorageOperations
+import org.catrobat.catroid.io.asynctask.ProjectExporter
+import org.catrobat.catroid.io.asynctask.saveProjectSerial
+import org.catrobat.catroid.utils.notifications.StatusBarNotificationManager
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito
+import java.io.File
+import java.io.IOException
+import java.util.Enumeration
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+
+@RunWith(AndroidJUnit4::class)
+class ProjectExporterTest {
+    @get:Rule
+    var runtimePermissionRule = GrantPermissionRule.grant(
+        Manifest.permission.WRITE_EXTERNAL_STORAGE,
+        Manifest.permission.READ_EXTERNAL_STORAGE
+    )
+
+    private lateinit var project: Project
+    private lateinit var contextMock: Context
+
+    @Before
+    fun setUp() {
+        project = Project(
+            ApplicationProvider.getApplicationContext(),
+            ProjectExporterTest::class.java.simpleName
+        )
+        ProjectManager.getInstance().currentProject = project
+        saveProjectSerial(project, ApplicationProvider.getApplicationContext())
+        val notificationManagerMock = Mockito.mock(NotificationManager::class.java)
+        contextMock = Mockito.mock(Context::class.java)
+        Mockito.`when`(contextMock.resources)
+            .thenReturn(ApplicationProvider.getApplicationContext<Context>().resources)
+        Mockito.`when`(contextMock.getSystemService(ArgumentMatchers.anyString()))
+            .thenReturn(notificationManagerMock)
+    }
+
+    @Test
+    fun exportProjectTest() {
+        createUndoCodeXmlFile()
+        val notificationManager = StatusBarNotificationManager(contextMock)
+        val notificationData = notificationManager.createSaveProjectToExternalMemoryNotification(
+            ApplicationProvider.getApplicationContext(),
+            project.name
+        )
+        ProjectExporter(
+            project.directory,
+            notificationData,
+            ApplicationProvider.getApplicationContext()
+        ).exportProjectToExternalStorage()
```

Comment:
Please use runBlocking to test if the thread was started :)
And afterward maybe we can make "exportProjectToExternalStorage" private ;)
```suggestion
runBlocking {
        ).exportProjectToExternalStorageAsync(this)
        }
```